src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -664,51 +668,63 @@ static int fmi3_xml_str_to_array_floatXX(fmi3_xml_parser_context_t* context, con
 
     assert(str);
 
-    /* create a copy that it's OK that strtok mutates */
+    /* Check first that the current function has implemented support for the type passed to this function.
+    *   Otherwise we will have errors further down where check if the parsing of scalars/arrays is OK.
+    */
+    if (!(fmi3_base_type_enu_is_int(primType->baseType) ||
+          fmi3_base_type_enu_is_float(primType->baseType))) {
+        assert(0);
+    }
+
+    /* Create a copy that it's OK that strtok mutates */
     strCopy = context->callbacks->malloc(strlen(str) + 1); /* plus one for null character */
     if (!strCopy) {
         JM_LOG_ERROR_NO_MEM();
         res = -1;
-        goto err1;
+        goto strCopy_malloc_error;
     }
     strncpy(strCopy, str, strlen(str) + 1);
 
-    /* allocate memory for the start values */
+    /* Allocate memory for the start values */
     vals = context->callbacks->malloc(nVals * primType->size); /* freed in fmi3_xml_clear_model_description */
     if (!vals) {
         JM_LOG_ERROR_NO_MEM();
         res = -1;
         goto err2;
     }
 
-    /* write start value(s) to correct type */
+    /* Write start value(s) to correct type */
     if (nVals == 1) { /* scalar */
-        if (fmi3_xml_str_to_floatXX(context, 1, vals, NULL, str, primType)) {
-            jm_log_error(context->callbacks, module, "Unable to parse to %s: %s", primType->name, str);
-            res = -1;
-            goto err2;
+        if ((fmi3_base_type_enu_is_int(primType->baseType) &&
+                fmi3_xml_str_to_intXX(context, 1, vals, NULL, str, primType))
+                ||
+            (fmi3_base_type_enu_is_float(primType->baseType) &&
+                fmi3_xml_str_to_floatXX(context, 1, vals, NULL, str, primType))) {
```

Comment:
We were going to discuss this on another place (and I think this is hard to read - you can break down large expressions to local variables if that increases readability)